basic-2/src/test/java/PageObject.java

```diff
@@ -0,0 +1,51 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public abstract class PageObject {
+		protected ChromeDriver driver;
+		protected String baseUrl;
+		protected static long ATTRIBUTE_TO_MATCH_TIMEOUT_SECONDS= 5;
+		
+		protected PageObject(ChromeDriver driver, String baseUrl) {
+			this.driver = driver;
+			this.baseUrl = baseUrl;
+
+			PageFactory.initElements(driver, this);
+		}
+		
+		protected WebElement getProductPageElement(String productElement) {
+			return driver.findElement(By.xpath("//div[@id='global-topnav']//a[@class = '"+productElement+"']"));
+		}
+		
+		protected WebElement getClassicPageElement(String productElement) {
+			return driver.findElement(By.xpath("//td[@class='borderBottom']/a[contains(@href, '"+productElement+"')]"));
+		}
+		
+		protected WebElement getSvtProProductPageElement(String productElement) {
+			return driver.findElement(By.xpath("//tbody//a[contains(text(), '"+productElement+"')]"));
+		}
+		
+		protected WebElement getSvtProProductElement(String productElement) {
+			return driver.findElement(By.xpath("//tbody//a[contains(text(), '"+productElement+"')]"));
+		}
+		
+		protected WebElement WebWaitTimeInProductListPage(String idName) {

```

Comment:
It is okay to have helper method like this in superclasses. However this method is not generalized enough as it is and the name does not accurately describe what the method does. This method actually waits for an element to be visible and is not related to the ProductsListPage as the current name implies. You should generalize it by accepting a By parameter rather than classname. Otherwise you would need to rename it WaitForElementToBeVisibleByClassName(String classname).

So, rename it to WaitForElementToBeVisible(By locator) and change the parameter on line 40 to type of By.